src/MSBuildBinLogQuery/Construction/GraphBuilder.cs

```diff
@@ -32,13 +30,38 @@ public void HandleEvents(params BuildEventArgs[] buildEvents)
         private void ProjectStarted(object sender, ProjectStartedEventArgs args)
         {
             var id = args.BuildEventContext.ProjectInstanceId;
-            Projects.GetOrAdd(id, new ProjectNode(id, args));
+            Projects.TryAdd(id, new ProjectNode(id, args));
```

Comment:
This is interesting. It's ok for now, but please add a TODO comment:

> A single project instance can be built concurrently with distinct EntryPointTargets. Those should be tracked as part of the overall ProjectNode.

Reply:
Done.